week-3/Homework/mandatory/4-eligible-students.js

```diff
@@ -1,27 +1,105 @@
+  
+ They also need pupils who would act as assistants. Unfortunately, the number
+ of seats is limited on the spaceship therefore they set the following algorithm
+ to select students:
+
+ Only students who have attended enough classes are eligible to join the research teams.
 
-  Create a function which:
+ Finish the "getEligibleStudents" function which:
   - Accepts an array which contains all the students' names and their attendance counts
-    (see tests to confirm how this data will be structured)
+    For example:
+    [
+      ["Hunor", 10],
+      ["Magor", 1]
+    ]
+    See that each student's information is stored separately as an embedded array.
+    This is also called as 2 dimensional array when you can find array inside of array.
+
   - Returns an array containing only the names of the who have attended AT LEAST 8 classes
+
+  
 */
 
-function eligibleStudents() {
+// Remember how to access to embedded arrays
+
+let twoDimensionalArray = [
+  ["cat", "dog"],
+  ["giraffe", "lion", "elephant"]
+];
+
+// Examples accessing to array element directly
+// Example 1
+let pets = twoDimensionalArray[0]; // this reads the first embedded array 
+console.log(pets[1]); // this prints "cat" to the console
+
+// Example 2
+// You can also read straight the element of an embedded array:
+console.log(twoDimensionalArray[1][2]);
+// This reads the second embedded array and then reads its last element and finally prints "elephant" to the console
+
+// Examples accessing to array element through array methods
+// Example 1
+let moreThanTwoArrays = twoDimensionalArray.filter(embeddedArray => embeddedArray.length > 2);
+console.log(moreThanTwoArrays);
+// This filter only keeps embededd arrays that have more than 3 elements
+
+// Example 2
+let arrayLengths = twoDimensionalArray.map(embeddedArray => embeddedArray.length);
+console.log(arrayLengths);
+// This gives back how many elements of each array have
+
+/* + + + + + + + */
+/* Now try the exercise */
+
+function getEligibleStudents(classes) {
+  let numOfClasses = classes.filter(item=> item[1] >= 8)
+ return numOfClasses.map(arr => arr[0]);
+}
+
+/*
+ 
+ Later leaders of Alpha planet decided to change the rule as below:
+
+ Only students whose name starts with the same letter of the name of the planet are eligible to join.
+
+ Implement the body of function called "getEligibleStudents2" which:
+  - Accepts an array in the same structure as before.
+  - Returns an array containing only the names that satisfies the new rule.
+  
+  Note:
+  Unfortunately, administrators messed up the letter casing of names, sometimes it starts with small letter.
 
+  Hint: To complete the function, search how to change text to lower or upper case by using string method.
+*/
+
+function getEligibleStudents2(names) {
+  let numOfClasses = names.filter(item=> item[0].startsWith("A") || item[0].startsWith("a"))
+  return numOfClasses.map(arr => arr[0]);
 }
```

Comment:
In this exercise, you did well and it is correct. Can you use what was said in the instructions.